src/sgame/components/OvermindComponent.cpp

```diff
@@ -72,35 +72,40 @@ Entity* OvermindComponent::FindTarget() {
 	return target;
 }
 
+/**
+ * Find target to attack, which means entity which is:
+ * 1: not on same team
+ * 2: in attack range
+ * 3: the lower health (raw health, armor not considered)
+ * 4: the closer
+ *
+ * a: candidate
+ * b: current target
+ * TODO: make them const (might require deeper changes)
+ **/
 bool OvermindComponent::CompareTargets(Entity& a, Entity& b) const {
+	team_t selfTeam = GetTeamComponent().Team();
 	team_t aTeam = G_Team(a.oldEnt);
 	team_t bTeam = G_Team(b.oldEnt);
 
-	// Prefer humans.
-	if (aTeam == TEAM_HUMANS && bTeam != TEAM_HUMANS) return true;
-	if (aTeam != TEAM_HUMANS && bTeam == TEAM_HUMANS) return false;
+	//don't attack friends or non-players
+	if ( G_OnSameTeam( selfTeam, aTeam ) ) return false;
+	if ( G_OnSameTeam( selfTeam, bTeam ) || !G_IsPlayableTeam( bTeam ) ) return true;
 
+	//check if one is out of range
 	float aDistance = G_Distance(entity.oldEnt, a.oldEnt);
-	float bDistance = G_Distance(entity.oldEnt, b.oldEnt);
-
-	// Rules for humans.
-	if (aTeam == TEAM_HUMANS && bTeam == TEAM_HUMANS) {
-		// Prefer the one in attack range.
-		if (aDistance < ATTACK_RANGE && bDistance > ATTACK_RANGE) return true;
-		if (aDistance > ATTACK_RANGE && bDistance < ATTACK_RANGE) return false;
+	if (aDistance > ATTACK_RANGE) return false;
 
-		HealthComponent* aHealthComponent = a.Get<HealthComponent>();
-		HealthComponent* bHealthComponent = b.Get<HealthComponent>();
+	float bDistance = G_Distance(entity.oldEnt, b.oldEnt);
+	if (bDistance > ATTACK_RANGE) return true;
 
-		// The overmind senses weakness!
-		if (aHealthComponent && bHealthComponent) {
-			if (aHealthComponent->HealthFraction() < bHealthComponent->HealthFraction()) return true;
-			if (aHealthComponent->HealthFraction() > bHealthComponent->HealthFraction()) return false;
-		}
-	}
+	// The overmind senses weakness! (but not armor, so is kind of a joke)
+	HealthComponent* aHealthComponent = a.Get<HealthComponent>();
+	HealthComponent* bHealthComponent = b.Get<HealthComponent>();
 
-	// Break tie by preferring smaller distance.
-	return aDistance < bDistance;
+	// same health: attack the closer one.
+	if (aHealthComponent == bHealthComponent) return aDistance < bDistance;
+	return aHealthComponent->HealthFraction() < bHealthComponent->HealthFraction();
```

Comment:
```suggestion
	// none have health, look at the closest one
	if (!aHealthComponent && !bHealthComponent) return aDistance < bDistance;
	// prefer looking at the one that can be hurt: you don't want to beat a dead horse
	if (!aHealthComponent) return false;
	if (!bHealthComponent) return true;
	// The overmind senses weakness!
	return aHealthComponent->HealthFraction() < bHealthComponent->HealthFraction();
```

Reply:
at that point, I think that targets without HealthComponent should not trigger calls to this function, I'll do some testing to see if an ASSERT() will make the code crash or not.